### PR TITLE
feat: adds neq, lt, lte, gt, and gte filter to stream

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

Feature that adds the `newly supported realtime filters](https://github.com/supabase/walrus/issues/9#issuecomment-1018927685) support for stream method. 
The list is
- neq
- lt
- lte
- gt
- gte

## What is the current behavior?

The above filters are not supported in stream method.

## What is the new behavior?

The filters above are supported

Fixes https://github.com/supabase-community/supabase-flutter/issues/91

### CODE REVIEW

File: lib/src/supabase_stream_builder.dart [hunk 1]
```diff
@@ -139,16 +253,19 @@ class SupabaseStreamBuilder extends Stream<SupabaseStreamEvent> {
 
   Future<void> _getStreamData() async {
     _streamData = [];
+    String? realtimeFilter;
+    if (_streamFilter != null) {
```
Comment (DanMossa) [inline lib/src/supabase_stream_builder.dart:257]: Good work!
I usually think it's fine to use a variable like this, but `streamFilter` is being used a lot that I think null checking would be useful. 


```suggestion
  Future<void> _getStreamData() async {
      _StreamPostgrestFilter? currentStreamFilter = _streamFilter;
    _streamData = [];
    String? realtimeFilter;
    if (currentStreamFilter != null) {
```

And then replace `_streamFilter` with `currentStreamFilter` and you get to remove all the `!` everywhere.

**Review by DanMossa — APPROVED** [review]

**Comment by JasonChiu-dev** [comment]
supabase
        .from('tables')
        .stream(primaryKey: ['id'])
        **.eq('projectId', projectId)
        .gte('datetime', '2022-11-01')
        .lt('datetime', t'2022-11-02')**
        .order('datetime', ascending: true);

Is it possible to apply multi-filters like the above sample??

**Comment by dshukertjr** [comment]
Hi @JasonChiu-dev 

At the moment, because of the limitation on our realtime server, applying multiple filters is not possible. Our team is constantly working on improving its capabilities, so this it might be possible in the long run, but cannot promise anything at the moment.

**Comment by JasonChiu-dev** [comment]
> Thank you for your reply. Hope to see this feature soon.
Jason

**Pull request merged**